### PR TITLE
core/txbuilder: turn ints into uint32s on signing path

### PR DESCRIPTION
**Pull request by tessr**
gRPC requires that we specify the int type; on the signing path we chose uint32s.

### CODE REVIEW

File: core/txbuilder/types.go [hunk 1]
```diff
@@ -29,24 +29,24 @@ type Template struct {
 	sigHasher *bc.SigHasher
 }
 
-func (t *Template) Hash(idx int) bc.Hash {
+func (t *Template) Hash(idx uint32) bc.Hash {
 	if t.sigHasher == nil {
 		t.sigHasher = bc.NewSigHasher(t.Transaction)
 	}
-	return t.sigHasher.Hash(idx)
+	return t.sigHasher.Hash(int(idx))
```
Comment (bobg) [inline core/txbuilder/types.go]: This is not a safe conversion. Maybe update `SigHasher.Hash` to take a `uint32`?

Comment (tessr) [inline core/txbuilder/types.go]: Eeek, thanks, I'll look into it.

**Comment by tessr** [comment]
This is all fixed up 💃

**Comment by tessr** [comment]
I hope

**Comment by bobg** [comment]
LGTM

**Pull request merged**